apduWrapper.go

```diff
@@ -170,15 +182,33 @@ func UnwrapResponseAPDU(channel uint16, pipe <-chan []byte, packetSize int) ([]b
 	var totalSize uint16
 	var done = false
 
+	// return values from DeserializePacket
+	var result []byte
+	var responseSize uint16
+	var err error
+
+	foundZeroSequence := false
+	isSequenceZero := false
+
 	for !done {
 		// Read next packet from the channel
 		buffer := <-pipe
 
-		result, responseSize, err := DeserializePacket(channel, buffer, sequenceIdx)
+		result, responseSize, isSequenceZero, err = DeserializePacket(channel, buffer, sequenceIdx) // this may fail if the wrong sequence arrives (espeically if left over all 0000 was in the buffer from the last tx)
+
+		// Recover from a known error condition:
+		// * Discard messages left over from previous exchange until isSequenceZero == true
+		if foundZeroSequence == false && isSequenceZero == false {
+			continue
+		}
+		foundZeroSequence = true
+
 		if err != nil {
 			return nil, err
 		}
```

Comment:
Should you first check if `err != nil` right after calling `DeserializePacket` ?

Reply:
updated